### ISSUE TITLE
Metadata of RobotFramework_TestsuitesManagement (2)

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/robotframework-testsuitesmanagement#103]
Beneath others the `RobotFramework_TestsuitesManagement` introduces the following metadata:

```
Set suite metadata 'version_sw' to value ''.
Set suite metadata 'version_hw' to value ''.
Set suite metadata 'version_test' to value ''.
```

Why? This would only makes sense if the `RobotFramework_TestsuitesManagement` also provides a mechanism to define values as metadata. I do not see such a mechanism. What I define within a parameter configuration file will be a global parameter, but not a metadata.

Might it be worth to think about:
```
  "params" : {
              "metadata": {
                            "version_sw" : "1.2.3",
                            "version_hw" : "4.5.6"
                        }
             }
```
?

I added the following lines in my robot file:
```
Metadata    my_own_metadata    my_own_metadata_value
Metadata    version_hw    metadata_version_hw
```

Outcome in log file:
`Set suite metadata 'version_hw' to value ''.`

I logged `SUITE_METADATA`. Outcome:

Properly `SUITE_METADATA` contains:
`'my_own_metadata': 'my_own_metadata_value'`

But `version_hw` is empty:
`'version_hw': ''`

Finally I deactivated the `testsuite_setup` of `RobotFramework_TestsuitesManagement`.

Now `SUITE_METADATA` contains:
`'my_own_metadata': 'my_own_metadata_value', 'version_hw': 'metadata_version_hw'`
like expected.

Seems that the computation order is not really well designed.

Either the `RobotFramework_TestsuitesManagement` provides own mechanisms to define metadata, or must not have the hand on values defined within robot files immediately.

**Comment by test-fullautomation** [comment]
Hi Holger,

my assumtions are:
* this three values are handed over by command line as dynamic values
* inside the robot file this three values are considered as metadata (constants)
* initialization with empty string makes sense as it indicates that no values 
   were handed over by command line.
* usecase is that all metadata will be provided by the execution jenkins job by means of command line argumets.
* this all need to be properly documented

@namsonx : please confirm above points.
@HolQue : please support to improve the documentation if required.

Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Thomas and Holger,

**- "This three values are handed over by command line as dynamic values"**
Not only are handed over by command but also can set in *** Settings *** in robot testsuite.

**- "Inside the robot file this three values are considered as metadata (constants)"**
Yes, we defined this three values as metadata.

**- "Initialization with empty string makes sense as it indicates that no values were handed over by command line."**
Yes, this is what we designed. But there is the small typo in code cause the issue Holger mentioned in this ticket. 
I will provide new pull-request to fix it.
    
- "Usecase is that all metadata will be provided by the execution jenkins job by means of command line argumets."
Yes

Thank you,
Son

**Comment by namsonx** [comment]
Hello @test-fullautomation,

I have just discussed with Holger, and we have agreed on the proposal below:

- These three values are able to handed over by command line as well as set in robot test suites script.
- In case both methods are provided, the values which are handed over via command line are using as higher priority.
- In case users don't set these values, it will be log empty as it is.

please give us your opinion regarding to our proposal above?

Thank you,
Son

**Comment by HolQue** [comment]
I am still trying to get an overview about the impact. Let's think in use cases:

* Use case 1: 'version_sw', 'version_hw' and 'version_test' and 'my_own_test_metadata' not defined in
command line and also not defined as 'Metadata' in '*** Settings ***'

* Use case 2: 'version_sw', 'version_hw' and 'version_test' and 'my_own_test_metadata' defined in command line, but not as 'Metadata' in '*** Settings ***'

* Use case 3: 'version_sw', 'version_hw' and 'version_test' and 'my_own_test_metadata' defined as 'Metadata' in '*** Settings ***', but not in command line

* Use case 4: 'version_sw', 'version_hw' and 'version_test' and 'my_own_test_metadata' defined in both: as 'Metadata' in '*** Settings ***' and in command line

* Use case 5: 'version_sw' defined in command line, 'version_hw' defined as 'Metadata' in '*** Settings ***', 'version_test' not defined, 'my_own_test_metadata' defined as 'Metadata' in '*** Settings ***'

What shall be for every use case the output in log file?

To avoid the things becoming too much complicated, I propose the following:

* The RobotFramework_TestsuitesManagement prints a table in log file, listing all available metadata values (that is the content of ${SUITE METADATA}), no matter from where the values came from.
* For every value (out of ['version_sw', 'version_hw', 'version_test']) that is not present in ${SUITE METADATA}, the RobotFramework_TestsuitesManagement adds a corresponding initialization before the content is logged.

What do you think?

**Comment by namsonx** [comment]
Hello Holger,

* "_The RobotFramework_TestsuitesManagement prints a table in log file, listing all available metadata values (that is the content of ${SUITE METADATA}), no matter from where the values came from._"
> I agree with you on this point.


* "For every value (out of ['version_sw', 'version_hw', 'version_test']) that is not present in ${SUITE METADATA}, the RobotFramework_TestsuitesManagement adds a corresponding initialization before the content is logged."
> What is a corresponding initialization for ['version_sw', 'version_hw', 'version_test']? As I understand 'version_sw' and 'version_hw' are software and hardware versions undertesting, they should be provided by users.

Thank you,
Son

**Comment by HolQue** [comment]
Hi Son,

it's mentioned here:

https://github.com/test-fullautomation/robotframework-testsuitesmanagement/issues/103#issuecomment-1426116318

"initialization with empty string makes sense as it indicates that no values
were handed over by command line."

Outcome:

```
Set suite metadata 'version_sw' to value ''.
Set suite metadata 'version_hw' to value ''.
Set suite metadata 'version_test' to value ''.
```

**Comment by namsonx** [comment]
Hello Holger,

I created the pull-request #126 for this issue.

Thank you,
Son

**Comment by test-fullautomation** [comment]
solved and already merged with #126
